zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/user/UpdateUserTest.java

```diff
@@ -79,4 +83,33 @@ public void shouldRejectUserUpdateCommandWhenUserKeyDoesNotExist() {
                 + username
                 + ", but a user with this username does not exist");
   }
+
+  @DisplayName("should set the key on follow up commands to the user key")
+  @Test
+  public void shouldSetTheKeyOnFollowUpCommandsToTheUserKey() {
+    final var username = UUID.randomUUID().toString();
+    final var userRecord =
+        ENGINE
+            .user()
+            .newUser(username)
+            .withName("Foo Bar")
+            .withEmail("foo@bar.com")
+            .withPassword("password")
+            .create();
+
+    ENGINE
+        .user()
+        .updateUser(userRecord.getKey())
+        .withUsername(userRecord.getValue().getUsername())
+        .withName("Bar Foo")
+        .withEmail("foo@bar.blah")
+        .withPassword("Foo Bar")
+        .update()
+        .getValue();
+
+    Assertions.assertThat(
+            RecordingExporter.userRecords(UserIntent.UPDATED).collect(Collectors.toList()))
+        .allMatch(record -> record.getKey() == userRecord.getKey());
```

Comment:
❌ We need to add a limit here. Otherwise this test will run for 5 seconds every time. I'd filter it on userKey, and then limit it to 1, as there can only be one UPDATED for this user in this test scenario.